route_following_plugin/src/route_following_plugin.cpp

```diff
@@ -92,16 +93,62 @@ void setManeuverLaneletIds(carma_planning_msgs::msg::Maneuver& mvr, lanelet::Id
     config_.lane_change_plugin_= declare_parameter<std::string>("lane_change_plugin", config_.lane_change_plugin_);
     config_.stop_and_wait_plugin_ = declare_parameter<std::string>("stop_and_wait_plugin", config_.stop_and_wait_plugin_);
     config_.lanefollow_planning_tactical_plugin_ = declare_parameter<std::string>("lane_following_plugin", config_.lanefollow_planning_tactical_plugin_);
-    config_.route_end_point_buffer_ = declare_parameter<double>("/guidance/route/destination_downtrack_range", config_.route_end_point_buffer_);
+    config_.route_end_point_buffer_ = declare_parameter<double>("guidance/route/destination_downtrack_range", config_.route_end_point_buffer_);
     config_.accel_limit_ = declare_parameter<double>("vehicle_acceleration_limit", config_.accel_limit_);
     config_.lateral_accel_limit_ = declare_parameter<double>("vehicle_lateral_accel_limit", config_.lateral_accel_limit_);
     config_.stopping_accel_limit_multiplier_ = declare_parameter<double>("stopping_accel_limit_multiplier", config_.stopping_accel_limit_multiplier_);
+<<<<<<< HEAD
+=======
     config_.vehicle_id = declare_parameter<std::string>("vehicle_id", config_.vehicle_id);
+>>>>>>> develop
     config_.min_maneuver_length_ = declare_parameter<double>("min_maneuver_length", config_.min_maneuver_length_);
   }
 
-    carma_ros2_utils::CallbackReturn RouteFollowingPlugin::on_configure_plugin()
+  carma_ros2_utils::CallbackReturn RouteFollowingPlugin::on_configure_plugin()
   {
+    config_ = Config();
+
+    get_parameter<double>("minimal_plan_duration", config_.min_plan_duration_);
+    get_parameter<std::string>("lane_change_plugin", config_.lane_change_plugin_);
+    get_parameter<std::string>("stop_and_wait_plugin", config_.stop_and_wait_plugin_);
+    get_parameter<std::string>("lane_following_plugin", config_.lanefollow_planning_tactical_plugin_);
+    get_parameter<double>("guidance/route/destination_downtrack_range", config_.route_end_point_buffer_);
+    get_parameter<double>("vehicle_acceleration_limit", config_.accel_limit_);
+    get_parameter<double>("vehicle_lateral_accel_limit", config_.lateral_accel_limit_);
+    get_parameter<double>("stopping_accel_limit_multiplier", config_.stopping_accel_limit_multiplier_);
+    get_parameter<double>("min_maneuver_length", config_.min_maneuver_length_);
+    
+    /* TODO make it different node parameter work
```

Comment:
What is going on with the changes in this file?

Reply:
I was trying to make the parameter "guidance/route/destination_downtrack_range" work as it works differently than ROS1 if it's a parameter from different node. However, this is not part of the scope of this story, so I created an issue for this, and reverted to original

Reply:
https://github.com/usdot-fhwa-stol/carma-platform/issues/1934